client/src/pages/ProfileListings/SearchBar/SearchBar.tsx

```diff
@@ -19,38 +19,75 @@ const StyledTextField = styled(TextField)(({ theme }) => ({
 }));
 
 interface State {
-  date: Date | null;
+  start: Date | null;
+  end: Date | null;
+  location: string;
+}
+
+interface Props {
+  start: string | null;
+  end: string | null;
   location: string | null;
 }
 
-const initialState: State = {
-  date: new Date(Date.now()),
-  location: null,
+const initialState = {
+  start: new Date(Date.now()),
+  end: new Date(Date.now()),
+  location: '',
+} as const;
+
+const formatInitialState = (from: string | null, to: string | null, location: string | null) => {
+  const curDate = new Date(Date.now()),
+    dateFrom = from ? new Date(parseInt(from)) : initialState.start,
+    dateTo = to ? new Date(parseInt(to)) : initialState.end,
+    start = dateFrom.getTime() >= curDate.getTime() ? dateFrom : curDate,
+    end = dateTo.getTime() >= curDate.getTime() && dateTo.getTime() >= start.getTime() ? dateTo : start;
+
+  return {
+    location: location ? location : initialState.location,
+    start,
+    end,
+  };
 };
 
-const SearchBar = () => {
+const SearchBar = ({ location, start, end }: Props) => {
   const classes = useStyles();
-  const [value, setValue] = useState<State>(initialState);
+  const curDate = new Date(Date.now());
```

Comment:
You just need to instantiate a `new Date()` here, and do not need the `Date.now()` part.